### PR TITLE
Fractal

### PR DESCRIPTION
**Pull request by shea786**

### CODE REVIEW

File: app/Transformers/AlbumsTransformer.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Transformers;
+
+use App\Album;
+use League\Fractal\TransformerAbstract;
+
+class AlbumsTransformer extends TransformerAbstract
```
Comment (szainmehdi) [inline app/Transformers/AlbumsTransformer.php]: Each of these `Transformer` classes should be singular. It's only job is to transform a single entity. 

`AlbumTransformer` for example.

File: app/Transformers/TracksTransformer.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    public function transform(Track $track)
+    {
+        return [
+            'id' => $track->id,
+            'slug' => $track->slug,
+            'album_id' => $track->album_id,
+            'name' => $track->name,
+            'file' => $track->file_path,
+            'link' => $track->mp3_link,
```
Comment (szainmehdi) [inline app/Transformers/TracksTransformer.php]: This should be `audio_file_url` maybe?

File: app/Transformers/TracksTransformer.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+     * @return array
+     */
+    public function transform(Track $track)
+    {
+        return [
+            'id' => $track->id,
+            'slug' => $track->slug,
+            'album_id' => $track->album_id,
+            'name' => $track->name,
+            'file' => $track->file_path,
```
Comment (szainmehdi) [inline app/Transformers/TracksTransformer.php]: What is `file_path`?

**Review by szainmehdi — CHANGES_REQUESTED** [review]
Also remember to update the `RecitersApiTest`

File: app/Transformers/TracksTransformer.php (hunk 3 above)
Comment (shea786) [reply app/Transformers/TracksTransformer.php]: so currently we have two options, either upload a mp3 or specify a url
file_path is the path to the uploaded file
mp3_link is the link to the external mp3 file

i dont know if we want to change this in the database level and change the names

**Review by szainmehdi — APPROVED** [review]

**Pull request merged**